src/EnergyPlus/MixedAir.cc

```diff
@@ -3279,6 +3275,11 @@ namespace MixedAir {
 						SetupEMSActuator( "Outdoor Air Controller", OAController( OAControllerLoop ).Name, "Air Mass Flow Rate", "[kg/s]", OAController( OAControllerLoop ).EMSOverrideOARate, OAController( OAControllerLoop ).EMSOARateValue );
 					}
 
+					if ( OAController( OAControllerLoop ).VentMechObjectNum > 0 && thisOASys > 0){
+						if ( !VentilationMechanical( OAController( OAControllerLoop ).VentMechObjectNum ).DCVFlag ){
+							AirLoopControlInfo( thisOASys ).AirLoopDCVFlag = false;
```

Comment:
thisOASys is the wrong index, it should be thisAirLoop
Also, the other two blocks with the new `if ( AirLoopZoneInfoSize >= AirLoopNum ){` should move into here and then this if won't be needed anymore.  This block here does not execute until NumPrimaryAirSys > 0 which means the airloop inputs have been processed and then the zone checks can proceed safely.  As it stand those checks won't happen at all if there's an ERV in the file.